src/Cardano/Wallet/Primitive/AddressDiscovery.hs

```diff
@@ -226,9 +225,13 @@ nextAddresses !key (AddressPoolGap !g) !cc !fromIx =
         (>= fromIx)
     newAddress = keyToAddress . deriveAddressPublicKey key cc
 
-newtype SeqState = SeqState (AddressPool, AddressPool)
+data SeqState = SeqState
+    { internalPool :: !AddressPool
+    , externalPool :: !AddressPool
+    }
```

Comment:
Defining named accessors to prevent mistake when dealing with the state.

Reply:
:100: